packages/squarev2-integration/src/SquareV2Form.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+    CardInstrument,
+    CheckoutSelectors,
+    CheckoutService,
+    PaymentMethod,
+} from '@bigcommerce/checkout-sdk';
+import classNames from 'classnames';
+import React, { FunctionComponent, useCallback, useEffect, useMemo, useState } from 'react';
+
+import {
+    CardInstrumentFieldset,
+    StoreInstrumentFieldset,
+} from '@bigcommerce/checkout/instrument-utils';
+import { usePaymentFormContext } from '@bigcommerce/checkout/payment-integration-api';
+import { LoadingOverlay } from '@bigcommerce/checkout/ui';
+
+export interface SquareV2FormProps {
+    checkoutService: CheckoutService;
+    checkoutState: CheckoutSelectors;
+    containerId: string;
+    deinitializePayment: () => Promise<void>;
+    initializePayment: () => Promise<void>;
+    method: PaymentMethod;
+}
+
+const SquareV2Form: FunctionComponent<SquareV2FormProps> = ({
+    checkoutService,
+    checkoutState,
+    containerId,
+    deinitializePayment,
+    initializePayment,
+    method,
+}) => {
+    const [isAddingNewCard, setIsAddingNewCard] = useState(false);
+    const [selectedInstrumentId, setSelectedInstrumentId] = useState<string | undefined>(undefined);
+
+    const { getCustomer, getInstruments } = checkoutState.data;
+    const isSignedIn = getCustomer()?.isGuest;
+    const isInstrumentFeatureAvailable = !isSignedIn && Boolean(method.config.isVaultingEnabled);
+
+    useEffect(() => {
+        if (isInstrumentFeatureAvailable) {
+            void checkoutService.loadInstruments();
+        }
+    }, [checkoutService, isInstrumentFeatureAvailable]);
+
+    const instruments = useMemo(() => getInstruments(method) || [], [getInstruments, method]);
+    const shouldShowInstrumentFieldset = isInstrumentFeatureAvailable && instruments.length > 0;
+    const shouldShowCreditCardFieldset = !shouldShowInstrumentFieldset || isAddingNewCard;
```

Comment:
I might recommend to check with someone:
do you need to render credit cards for cases, when the customer has untrusted shipping address for the instruments?

This is a part of extra security to avoid fraud when paying with stored cards

Reply:
Checked with the team, it seems our project requirements do not include handling untrusted shipping addresses.